qiskit/primitives/statevector_sampler.py

```diff
@@ -0,0 +1,225 @@
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Statevector Sampler class
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit import ClassicalRegister, QiskitError, QuantumCircuit
+from qiskit.circuit import ControlFlowOp
+from qiskit.quantum_info import Statevector
+
+from .base import BaseSamplerV2
+from .base.validation import _has_measure
+from .containers import (
+    BasePrimitiveOptions,
+    BitArray,
+    PrimitiveResult,
+    PubResult,
+    SamplerPub,
+    SamplerPubLike,
+    make_data_bin,
+)
+from .containers.bit_array import _min_num_bytes
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class Options(BasePrimitiveOptions):
+    """Options for execution."""
+
+    shots: int = 100  # TODO: discuss the default number of shots
+    seed: Optional[Union[int, np.random.Generator]] = None
+
+
+@dataclass
+class _MeasureInfo:
+    creg_name: str
+    num_bits: int
+    num_bytes: int
+    qreg_indices: List[int]
+
+
+class Sampler(BaseSamplerV2):
+    """
+    Simple implementation of :class:`BaseSamplerV2` with Statevector.
+
+    :Options:
+
+        - **shots** (int) --
+          The number of shots.
+
+        - **seed** (np.random.Generator or int) --
+          Set a fixed seed or generator for the normal distribution.
+    """
+
+    _options_class = Options
+    options: Options
+
+    def __init__(self, *, options: Optional[BasePrimitiveOptions] = None):
+        """
+        Args:
+            options: Options including shots, seed.
+        """
+        if options is None:
+            self.options = Options()
+        elif isinstance(options, BasePrimitiveOptions):
+            self.options = options
+        else:
+            self.options = Options(**options)
+
+    def run(
+        self, pubs: Iterable[SamplerPubLike], shots: int | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        job: PrimitiveJob[PubResult] = PrimitiveJob(self._run, pubs, shots)
+        job._submit()
+        return job
+
+    def _run(
+        self, pubs: Iterable[SamplerPubLike], shots: int | None = None
+    ) -> PrimitiveResult[PrimitiveResult[PubResult]]:
+        if shots is None:
+            shots = self.options.shots
+        coerced_pubs = [SamplerPub.coerce(pub, shots) for pub in pubs]
+        for pub in coerced_pubs:
+            pub.validate()
+
+        results = []
+        for pub in coerced_pubs:
+            circuit, qargs, meas_info = _preprocess_circuit(pub.circuit)
+            bound_circuits = pub.parameter_values.bind_all(circuit)
+            arrays = {
+                item.creg_name: np.zeros(
+                    bound_circuits.shape + (shots, item.num_bytes), dtype=np.uint8
```

Comment:
In this loop, shots should be taken from the pub.

Reply:
Thanks. I updated it to use `pub.shots`.